gcloud-java-dns/src/main/java/com/google/gcloud/spi/DefaultDnsRpc.java

```diff
@@ -0,0 +1,192 @@
+    try {
+      ManagedZonesListResponse zoneList = dns.managedZones().list(this.options.projectId())
+          .setFields(FIELDS.getString(options))
+          .setMaxResults(PAGE_SIZE.getInt(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .execute();
+      return Tuple.<String, Iterable<ManagedZone>>of(zoneList.getNextPageToken(),
+          zoneList.getManagedZones());
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public boolean deleteZone(String zoneName) throws DnsException {
+    try {
+      dns.managedZones().delete(this.options.projectId(), zoneName).execute();
+      return true;
+    } catch (IOException ex) {
+      DnsException serviceException = translate(ex);
+      if (serviceException.code() == HTTP_NOT_FOUND) {
+        return false;
+      }
+      throw serviceException;
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<ResourceRecordSet>> listDnsRecords(String zoneName,
+      Map<Option, ?> options) throws DnsException {
+    // options are fields, page token, dns name, type
+    try {
+      ResourceRecordSetsListResponse response = dns.resourceRecordSets()
+          .list(this.options.projectId(), zoneName)
+          .setFields(FIELDS.getString(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .setMaxResults(PAGE_SIZE.getInt(options))
+          .setName(DNS_NAME.getString(options))
+          .setType(DNS_TYPE.getString(options))
+          .execute();
+      return Tuple.<String, Iterable<ResourceRecordSet>>of(response.getNextPageToken(),
+          response.getRrsets());
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Project getProject(Map<Option, ?> options) throws DnsException {
+    try {
+      return dns.projects().get(this.options.projectId())
+          .setFields(FIELDS.getString(options)).execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Change applyChangeRequest(String zoneName, Change changeRequest, Map<Option, ?> options)
+      throws DnsException {
+    try {
+      return dns.changes().create(this.options.projectId(), zoneName, changeRequest)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Change getChangeRequest(String zoneName, String changeRequestId, Map<Option, ?> options)
+      throws DnsException {
+    try {
+      return dns.changes().get(this.options.projectId(), zoneName, changeRequestId)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      DnsException serviceException = translate(ex);
+      if (serviceException.code() == HTTP_NOT_FOUND) {
+        // message format "The 'parameters.changeId' resource named '4' does not exist."
+        if (serviceException.getMessage().contains("changeId")) {
```

Comment:
Re error-codes, can we add that to the list. There should be a better way to detect that (e.g. different reason).